python/ray/serve/_private/application_state.py

```diff
@@ -1119,13 +1150,19 @@ def deploy_apps(self, name_to_deployment_args: Dict[str, List[Dict]]) -> None:
                 # against during this batch operation.
                 live_route_prefixes[deploy_app_prefix] = name
 
+            application_args = name_to_application_args.get(
+                name, ApplicationArgsProto(external_scaler_enabled=False)
+            )
```

Comment:
when do we expect `name_to_application_args` to not contain `ApplicationArgsProto` for a app? Doesn't the caller always pass it? If the caller is not passing it we should enforce it. wdyt?